python/ray/_raylet.pyx

```diff
@@ -1196,36 +1196,43 @@ async def execute_streaming_generator_async(
 
     Args:
         context: The context to execute streaming generator.
+        coroutine_complete_event: The asyncio.Event to notify the
```

Comment:
this and the type hint say asyncio.Event but a threading.Event is passed in

Reply:
ah I tried with non-blocking appraoch, and it didn't work, and I changed to blocking way. Let's just go with https://github.com/ray-project/ray/pull/40122 as it makes more sense to me..